rust/native/src/closefds.rs

```diff
@@ -0,0 +1,196 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! closefds is a library that provides support for setting the `FD_CLOEXEC` flag on all open
+//! file descriptors after `fork()` and before `exec()` on UNIX-like systems.
```

Comment:
`closefds` seems to be [available on crates.io](https://crates.io/crates/closefds), so I'm unclear on why you'd import the source directly into the project instead.

Reply:
it has a few issues, I'll give some details in the split.